app/models/bulk_download.rb

```diff
@@ -23,6 +23,47 @@ class BulkDownload < ApplicationRecord
   attr_accessor :params
 
   validates :status, presence: true, inclusion: { in: [STATUS_WAITING, STATUS_RUNNING, STATUS_ERROR, STATUS_SUCCESS] }
+  validate :params_checks
+
+  def params_checks
```

Comment:
The "rails way" and leading to simpler, easy to test functions would be to validate each parameter independently.

Reply:
Discussed offline.